qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -1155,23 +1155,49 @@ def _noncommutation_graph(self, qubit_wise):
         # results from one triangle to avoid symmetric duplications.
         return list(zip(*np.where(np.triu(adjacency_mat, k=1))))
 
-    def _create_graph(self, qubit_wise):
-        """Transform measurement operator grouping problem into graph coloring problem
+    def noncommutation_graph(self, qubit_wise: bool) -> rx.PyGraph:
+        """Create the non-commutation graph of this PauliList.
+
+        This transforms the measurement operator grouping problem into graph coloring problem. The
+        constructed graph contains one node for each Pauli. The nodes will be connecting for any two
+        Pauli terms that do _not_ commute.
 
         Args:
             qubit_wise (bool): whether the commutation rule is applied to the whole operator,
                 or on a per-qubit basis.
 
         Returns:
-            rustworkx.PyGraph: A class of undirected graphs
+            rustworkx.PyGraph: the non-commutation graph with nodes for each Pauli and edges
+                indicating a non-commutation relation.
         """
-
         edges = self._noncommutation_graph(qubit_wise)
         graph = rx.PyGraph()
         graph.add_nodes_from(range(self.size))
         graph.add_edges_from_no_data(edges)
         return graph
```

Comment:
We should probably comment on the node and edge data of the return value, which leads to a second question: currently the node value is indices into this `PauliList`, but a second potential choice is attaching `Pauli` objects themselves to the nodes.

Providing indices is better from a memory perspective, providing individual `Pauli` objects would be more convenient (a single `Pauli` involves two Numpy arrays and a bunch of extra data!).  Do you have any thoughts on which would be preferable for you?

Reply:
I think having just the index would be good enough for me. I had not thought about this much yet tbh though.
I can see an argument being made for including the full `Pauli` object, but since this graph can become quite large when dealing with large operators, maybe we should minimize the memory footprint for the time being.

I am open to other inputs though :+1:

Reply:
Yeah, just the index feels more right to me - if nothing else, it lets the code be more easily shared between `PauliList` and `SparsePauliOp`, where the latter would need the index to retrieve the coefficient.  It would be good to comment on what the node data in the returned `PyGraph` is in the docstrings, I think.

Reply:
I have added this to the docs. Leaving this thread unresolved in case anyone else has more input on this.